File: app/src/main/java/android/ufrpe/com/explore/UserList.java
package android.ufrpe.com.explore;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.widget.Toast;

import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;

import java.util.ArrayList;
import java.util.List;

public class UserList extends AppCompatActivity {
    private RecyclerView mRecyclerView;
    private ImageAdapter mAdapter;

    private DatabaseReference mDataBaseRef;
    private List<User> mUser;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_user_list);

        mRecyclerView = (RecyclerView) findViewById(R.id.recycler_view);
        mRecyclerView.setHasFixedSize(true);
        mRecyclerView.setLayoutManager(new LinearLayoutManager(this));

        mUser = new ArrayList<>();

        mDataBaseRef = FirebaseDatabase.getInstance().getReference("users");
        mDataBaseRef.addValueEventListener(new ValueEventListener() {
            @Override
            public void onDataChange(DataSnapshot dataSnapshot) {
                for (DataSnapshot postSnapShot: dataSnapshot.getChildren()){
                    User user = postSnapShot.getValue(User.class);
                    mUser.add(user);
                }

                mAdapter = new ImageAdapter(UserList.this,mUser);
                mRecyclerView.setAdapter(mAdapter);

            }

            @Override
            public void onCancelled(DatabaseError databaseError) {
                Toast.makeText(UserList.this,databaseError.getMessage(),Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: app/src/main/java/android/ufrpe/com/explore/UserList.kt
package android.ufrpe.com.explore

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.widget.Toast

import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.database.ValueEventListener

import java.util.ArrayList

class UserList : AppCompatActivity() {
    private var mRecyclerView: RecyclerView? = null
    private var mAdapter: ImageAdapter? = null

    private var mDataBaseRef: DatabaseReference? = null
    private var mUser: MutableList<User>? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_user_list)

        mRecyclerView = findViewById(R.id.recycler_view) as RecyclerView
        mRecyclerView!!.setHasFixedSize(true)
        mRecyclerView!!.layoutManager = LinearLayoutManager(this)

        mUser = ArrayList<User>()

        mDataBaseRef = FirebaseDatabase.getInstance().getReference("users")
        mDataBaseRef!!.addValueEventListener(object : ValueEventListener {
            override fun onDataChange(dataSnapshot: DataSnapshot) {
                for (postSnapShot in dataSnapshot.children) {
                    val user = postSnapShot.getValue<User>(User::class.java)
                    mUser!!.add(user!!)
                }

                mAdapter = ImageAdapter(this@UserList, mUser!!)
                mRecyclerView!!.adapter = mAdapter

            }

            override fun onCancelled(databaseError: DatabaseError) {
                Toast.makeText(this@UserList, databaseError.message, Toast.LENGTH_SHORT).show()
            }
        })


    }


}
